### DISCUSSION TITLE
Constraint setup problem

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#560] [General]
![image](https://github.com/mdolab/dafoam/assets/115110434/3f3dd6d8-435d-40d0-b439-4972da7d2cbc)
![image](https://github.com/mdolab/dafoam/assets/115110434/2a84345d-4e49-4551-87dc-d387d8e59059)

Hello Professor He, I'm seeking guidance on addressing a challenge in my rotor blade optimization project. I've meticulously defined all the points within the rotor blade, yet I'm uncertain about the origin of the issue. Your assistance in pinpointing and resolving this problem would be invaluable. Currently, my focus is on optimizing the rotor blade to enhance lift and minimize drag. I appreciate any insights or suggestions you can provide. I am new here so 请您多多关照

[runScript.py.txt](https://github.com/mdolab/dafoam/files/13920159/runScript.py.txt)

**Comment by friedenhe** [comment]
Defining the leList and teList for a twisted blade is a bit tricky. The error you saw is caused by (1) the leList/teList are not completely within the blade or (2) when pyGeo tries to connect leList and teList to form a 2D point cloud (for computing blade thickness at these points), some points are outside the blade. To avoid this, I suggest you define your leList and teList well within the blade, i.e., far away from the actual leading and trailing edges. This can avoid the above (1) and (2) issues. If this work, you can gradually move leList and teList towards the actual leading and trailing edges.
